components/settings/Integrations.tsx

```diff
@@ -0,0 +1,169 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Button, InlineNotification, Modal } from '../common'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import { useSelector } from 'react-redux'
+import { RootState } from '../../store'
+import { newIslandConfig } from '../../lib/settings'
+import {
+  getPhoneIslandToken,
+  phoneIslandTokenCheck,
+  removePhoneIslandToken,
+} from '../../services/authentication'
+
+export const Integrations = () => {
+  const [copied, setCopied] = useState<boolean>(false)
+  const [config, setConfig] = useState<string>('')
+  const currentUser = useSelector((state: RootState) => state.user)
+  const webRTCExtension = currentUser.endpoints.extension.filter((el) => el.type === 'webrtc')[0]
+  const [showMondal, setShowMondal] = useState(false)
+  const cancelButtonRef: RefObject<HTMLButtonElement> = createRef()
+  const [tokenExists, setTokenExists] = useState<boolean>(false)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const newConfig = async () => {
+    setLoading(true)
+    const data = await getPhoneIslandToken()
+    if (data.token) {
+      const config = newIslandConfig({
+        // @ts-ignore
+        hostname: window.CONFIG.API_ENDPOINT,
+        username: data.username,
+        auth_token: data.token,
+        sip_exten: webRTCExtension.id,
+        sip_secret: webRTCExtension.secret,
+      })
+      setConfig(config)
+      setCopied(false)
+      setTokenExists(true)
+    }
+    setLoading(false)
+  }
+
+  const removeConfig = async () => {
+    const removed = await removePhoneIslandToken()
+    if (removed) {
+      setConfig('')
+      setTokenExists(false)
+      setShowMondal(false)
+    }
+  }
+
+  useEffect(() => {
+    const checkPhoneIslandToken = async () => {
+      const data = await phoneIslandTokenCheck()
+      if (data.exists) setTokenExists(true)
+    }
+    checkPhoneIslandToken()
+  }, [])
+
+  return (
+    <>
+      {/* The Integration section */}
+      <section aria-labelledby='phone-configuration-heading'>
+        <div className='shadow sm:overflow-hidden w-full dark:bg-gray-900'>
+          <div className='bg-white py-6 px-4 sm:p-6 w-full dark:bg-gray-900'>
+            <div>
+              <h2 className='text-lg font-medium leading-6 text-gray-900 dark:text-gray-100 mb-6'>
+                Integrations
+              </h2>
+            </div>
+            <div>
+              <h4
+                id='phone-configuration-heading'
+                className='text-sm font-medium leading-6 text-gray-900 dark:text-gray-100'
+              >
+                External phone configuration
+              </h4>
+              <p className='mt-1 text-sm text-gray-500 dark:text-gray-400'>
+                Phone configuration can be imported into other applications until integrations are
+                ready to work with this one. Creating a new configuration will invalidate the
```

Comment:
If the user clicks "Create" button the current configuration is immediately invalidated? Maybe it's better to show a confirmation modal with this description: "If you generate a new phone widget configuration, any phone widget in your web applications must be reconfigured with the new string"

Reply:
I disabled the creation button instead.